lib/trino-filesystem/src/main/java/io/trino/filesystem/Location.java

```diff
@@ -209,6 +201,33 @@ public Location appendPath(String newPathElement)
         return withPath(location + newPathElement, path + newPathElement);
     }
 
+    /**
+     * Creates a new location by appending the given suffix to the current path.
```

Comment:
So a "suffix" is a file ending, like ".zip".  If so, add another line to this comment saying something like "this is commonly used to add a file name extension like .zip"

Reply:
It can be anything. We use it in the test to force adding a trailing slash. I'll add a comment explaining how this should be used and when to use the other one.